commit 1cf50fa80a674a5a427df733ac58eb162b7aadba
Author: dyep <dyep49@gmail.com>
Date:   2019-04-26T08:33:04+00:00 (6 months ago)
    
    TRITON-1299 need logarchiver-agent service to run along side hermes-actor
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/actor/actor.js b/actor/actor.js
index 6e36b9b..39e01bf 100644
--- a/actor/actor.js
+++ b/actor/actor.js
@@ -593,7 +593,8 @@ resched()
 function
 main()
 {
-	lib_utils.create_logger(GS, 'hermes-actor');
+	var service_name = GS.gs_sfmri.split('/').reverse()[0];
+	lib_utils.create_logger(GS, service_name);
 
 	mod_vasync.pipeline({
 		funcs: [
diff --git a/hermes.js b/hermes.js
index 9e67b00..34fbd8b 100644
--- a/hermes.js
+++ b/hermes.js
@@ -112,6 +112,10 @@ read_config()
 		GS.gs_log.level(process.env.LOG_LEVEL || cfg.log_level ||
 		    mod_bunyan.INFO);
 
+		if (cfg.service_name && cfg.service_name === 'sdc') {
+			cfg.service_name = 'hermes';
+		}
+
 		/*
 		 * Validate the configuration before returning it:
 		 */
@@ -156,6 +160,18 @@ validate_config(cfg)
 		}
 	}
 
+	if (!cfg.service_name) {
+		GS.gs_log.info('configuration missing "service_name"');
+		return(false);
+	}
+
+	var valid_service = cfg.service_name === 'hermes' ||
+		cfg.service_name === 'logarchiver';
+
+	mod_assert.ok(valid_service, 'service_name must be "hermes" or ' +
+		'"logarchiver"');
+
+
 	var check_number = function check_number(name) {
 		if (!cfg[name] || typeof (cfg[name]) !== 'number' ||
 		    isNaN(cfg[name])) {
@@ -194,6 +210,7 @@ bootstrap_server(server)
 
 	var script = GS.gs_scriptmgr.load('bootstrap.ksh', {
 		ENDPOINT: GS.gs_config.admin_ip + ':' + GS.gs_config.port,
+		AGENT_NAME: GS.gs_config.agent_name,
 		SMF_REVISION: 'HERMES-1'
 	});
 	server.execute([], {}, script, function (err, res) {
@@ -351,10 +368,12 @@ main()
 	GS.gs_servermgr.deployed_version(GS.gs_tarstamp);
 
 	log.debug('starting http server');
+	GS.gs_config.agent_name = GS.gs_config.service_name === 'hermes' ?
+		'hermes-actor' : 'logarchiver-agent';
 	GS.gs_httpserver = new lib_httpserver.HttpServer(log.child({
 		component: 'HttpServer'
 	}), GS.gs_config.admin_ip, GS.gs_config.port, GS.gs_tarstamp,
-	    GS.gs_scriptmgr);
+	    GS.gs_scriptmgr, GS.gs_config.agent_name);
 
 	GS.gs_httpserver.on('shed', function (shed) {
 		GS.gs_servermgr.accept(shed);
diff --git a/lib/httpserver.js b/lib/httpserver.js
index d63d4a4..08c781f 100644
--- a/lib/httpserver.js
+++ b/lib/httpserver.js
@@ -22,7 +22,7 @@ var WATERSHED = new mod_watershed.Watershed();
 
 
 function
-HttpServer(log, ip, port, tarstamp, scriptmgr)
+HttpServer(log, ip, port, tarstamp, scriptmgr, agent_name)
 {
 	var self = this;
 	mod_events.EventEmitter.call(self);
@@ -32,12 +32,14 @@ HttpServer(log, ip, port, tarstamp, scriptmgr)
 	mod_assert.number(port, 'port');
 	mod_assert.string(tarstamp, 'tarstamp');
 	mod_assert.object(scriptmgr, 'scriptmgr');
+	mod_assert.string(agent_name, 'agent_name');
 
 	self.hs_log = log;
 	self.hs_ip = ip;
 	self.hs_port = port;
 	self.hs_tarstamp = tarstamp;
 	self.hs_scriptmgr = scriptmgr;
+	self.service_name = agent_name;
 
 	setImmediate(function () {
 		self._init();
@@ -90,8 +92,10 @@ _init()
 	};
 
 	var get_script = function (filename, req, res, next) {
+
 		var script = self.hs_scriptmgr.load(filename, {
 			ENDPOINT: self.hs_ip + ':' + self.hs_port,
+			AGENT_NAME: self.agent_name,
 			VERSION: self.hs_tarstamp,
 			SMF_REVISION: 'HERMES-1'
 		});
diff --git a/package.json b/package.json
index f0b0356..191661a 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "hermes",
-  "version": "0.4.1",
+  "version": "0.5.0",
   "description": "Centralised tool to upload SDC logs to Manta",
   "main": "hermes.js",
   "author": "Joshua M. Clulow <jmc@joyent.com>",
diff --git a/sapi_manifests/hermes/template b/sapi_manifests/hermes/template
index 1142cee..c759940 100644
--- a/sapi_manifests/hermes/template
+++ b/sapi_manifests/hermes/template
@@ -20,5 +20,6 @@
     "key_id": "{{{SDC_KEY_ID}}}",
     "connect_timeout": "{{{HERMES_CONNECT_TIMEOUT}}}"
   },
-  "max_concurrent_bootstraps": 32
+  "max_concurrent_bootstraps": 32,
+  "service_name": "{{{SERVICE_NAME}}}"
 }
diff --git a/scripts/actor.ksh b/scripts/actor.ksh
index dc4bb68..59ca001 100644
--- a/scripts/actor.ksh
+++ b/scripts/actor.ksh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 set -o errexit
@@ -15,19 +15,18 @@ set -o pipefail
 . /lib/svc/share/smf_include.sh
 
 CURL=/usr/bin/curl
-GUNZIP=/usr/bin/gunzip
 TAR=/usr/bin/tar
 DIGEST=/usr/bin/digest
 SVCCFG=/usr/sbin/svccfg
 SVCADM=/usr/sbin/svcadm
 
-if [[ $SMF_FMRI != "svc:/smartdc/hermes-actor:default" ]]; then
+if [[ $SMF_FMRI != "svc:/smartdc/%%AGENT_NAME%%:default" ]]; then
 	printf "ERROR: not running under correct SMF service\n" >&2
 	exit $SMF_EXIT_ERR_NOSMF
 fi
 
-ACTOR_DIR="/opt/smartdc/hermes-actor"
-TMPFILE="/tmp/.hermes-actor.$$.tar.gz"
+ACTOR_DIR="/opt/smartdc/%%AGENT_NAME%%"
+TMPFILE="/tmp/.%%AGENT_NAME%%.$$.tar.gz"
 DEPLOY_DIR="${ACTOR_DIR}/deploy"
 VERFILE="${DEPLOY_DIR}/.version"
 
diff --git a/scripts/actor.xml b/scripts/actor.xml
index 59bbee4..642c60e 100644
--- a/scripts/actor.xml
+++ b/scripts/actor.xml
@@ -7,11 +7,11 @@
 -->
 
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright (c) 2019, Joyent, Inc.
 -->
 
-<service_bundle type="manifest" name="smartdc-hermes-actor">
-  <service name="smartdc/hermes-actor" type="service" version="1">
+<service_bundle type="manifest" name="smartdc-%%AGENT_NAME%%">
+  <service name="smartdc/%%AGENT_NAME%%" type="service" version="1">
 
     <create_default_instance enabled="false"/>
     <single_instance/>
@@ -20,7 +20,7 @@
       <service_fmri value="svc:/milestone/multi-user:default"/>
     </dependency>
 
-    <exec_method type="method" name="start" exec="/opt/smartdc/hermes-actor/method.ksh" timeout_seconds="600">
+    <exec_method type="method" name="start" exec="/opt/smartdc/%%AGENT_NAME%%/method.ksh" timeout_seconds="600">
       <method_context working_directory="/tmp"/>
     </exec_method>
 
diff --git a/scripts/bootstrap.ksh b/scripts/bootstrap.ksh
index 109e21b..f409798 100644
--- a/scripts/bootstrap.ksh
+++ b/scripts/bootstrap.ksh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 #
@@ -22,8 +22,8 @@ SVCCFG=/usr/sbin/svccfg
 SVCPROP=/usr/bin/svcprop
 CURL=/usr/bin/curl
 
-ACTOR_DIR="/opt/smartdc/hermes-actor"
-FMRI="svc:/smartdc/hermes-actor:default"
+ACTOR_DIR="/opt/smartdc/%%AGENT_NAME%%"
+FMRI="svc:/smartdc/%%AGENT_NAME%%:default"
 
 SERVER="%%ENDPOINT%%"
 SMF_REVISION="%%SMF_REVISION%%"
